### ISSUE TITLE
Does not exist error

### ISSUE THREAD

**Issue by mightymouse2045** [NixOS/nixpkgs#32808]
## Issue description
Running nix-shell produces 'Does not exist error'

Hi I am trying to build Cardano-sl and it is failing to load the nix-shell with:

#nix-shell
build input /nix/store/q8k7r5zbf03ydrpm7imfzds5b2nmdf0a-zlib-1.2.11-dev "......." /nix/store/15axh70yqanvivgi69ns8sx7cjydqdx3-pkg-config-0.29.2 does not exist

There are quite a few more paths included in the above and they all exist as they should. But no matter what I do, delete and start from scratch, different branch etc, it gives me the same error. 

Is there anyway I can troubleshoot this further to find out what's going on, or hopefully something obvious/easy that anyone knows to work around this?

### Steps to reproduce
Download any package and run 'nix-shell' produces the same result

## Technical details
these paths will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/fxkr6d8ps6dd0qiy0c19kdjg96psindz-nix-info
fetching path ‘/nix/store/fxkr6d8ps6dd0qiy0c19kdjg96psindz-nix-info’...

*** Downloading ‘https://cache.nixos.org/nar/1v76qd1lbps54ixpvir7qph3dcq79c8kmwyfa5ngq7g2pg7iv4ls.nar.xz’ to ‘/nix/store/fxkr6d8ps6dd0qiy0c19kdjg96psindz-nix-info’...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1644  100  1644    0     0    822      0  0:00:02  0:00:02 --:--:--   679

 - system: `"x86_64-linux"`
 - host os: `Linux 4.13.9-300.fc27.x86_64, Fedora, 27 (Workstation Edition)`
 - multi-user?: `no`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(gh): `"nixpkgs-18.03pre122585.b212125b541"`
 - nixpkgs: `/home/gh/.nix-defexpr/channels/nixpkgs`

**Comment by edolstra** [comment]
Please provide detailed steps to reproduce and what the expected result is. Don't say "Download any package and run 'nix-shell'" because I don't even know what that's supposed to *do*.

**Comment by mightymouse2045** [comment]
I am trying to build cardano-sl a nix package - the instructions are run 'nix-shell' from within the folder I have downloaded. There is a default.nix file in there, it goes off and downloads everything and then fails with the error above - 'does not exist'.

I have just downloaded nixos and run it up in virtualbox, and voila everything works as expected and I get into a nix-shell prompt "[nix-shell:~/cardano-sl]$"

In Fedora 27 with a fresh install of nix I cannot get into the nix-shell it fails with the above error. This is my first time using nix.

I have tried running nix-shell with other nix packages and it fails the same way - so personally I think it is an issue with Fedora in some way, but I have not clue how to troubleshoot this at all,
